onnx/backend/test/case/node/__init__.py

```diff
@@ -128,31 +128,45 @@ def rename_helper(internal_name: str) -> Any:
 
 def function_testcase_helper(
     node: NodeProto, input_types: List[TypeProto], name: str
-) -> Tuple[List[NodeProto], List[OperatorSetIdProto]]:
+) -> Tuple[List[List[NodeProto]], List[Any], int]:
     test_op = node.op_type
     op_prefix = test_op + "_" + name + "_expanded_function_"
     schema = onnx.defs.get_schema(test_op, node.domain)
 
-    if schema.has_function:  # type: ignore
-        function_proto = schema.function_body  # type: ignore
-    elif schema.has_context_dependent_function:  # type: ignore
-        function_proto_str = schema.get_context_dependent_function(  # type: ignore
-            node.SerializeToString(), [t.SerializeToString() for t in input_types]
+    # an op schema may have several functions, each for one opset version
+    # opset versions include the op's since_version and other opset versions
+    # if it is needed to define the op for a opset version other than the op's since_version.
+    function_protos = []
+    for opset_version in schema.function_opset_versions:  # type: ignore
+        function_proto_str = schema.get_function_with_opset_version(opset_version)  # type: ignore
+        function_proto = FunctionProto()
+        function_proto.ParseFromString(function_proto_str)
+        function_protos.append(function_proto)
+    for opset_version in schema.context_dependent_function_opset_versions:  # type: ignore
+        function_proto_str = schema.get_context_dependent_function_with_opset_version(  # type: ignore
+            opset_version,
+            node.SerializeToString(),
+            [t.SerializeToString() for t in input_types],
         )
         function_proto = FunctionProto()
         function_proto.ParseFromString(function_proto_str)
-    else:
-        return [], []
+        function_protos.append(function_proto)
+
+    node_lists = []
+    function_proto_opset_imports = []
+    for function_proto in function_protos:
+        for attr in schema.attributes:
+            if attr in [a.name for a in node.attribute]:
+                continue
+            if schema.attributes[attr].default_value:
+                node.attribute.extend([schema.attributes[attr].default_value])
 
-    for attr in schema.attributes:
-        if attr in [a.name for a in node.attribute]:
-            continue
-        if schema.attributes[attr].default_value:
-            node.attribute.extend([schema.attributes[attr].default_value])
+        # function_proto.attributes
+        node_list = function_expand_helper(node, function_proto, op_prefix)
+        node_lists.append(node_list)
+        function_proto_opset_imports.append(function_proto.opset_import)
```

Comment:
minor nit: it may be cleaner to do:
```py
   expanded_tests.append(node_list, function_proto.opset_import)
```
That is, construct and return `zip(node_lists, function_proto_opset_imports)` directly.